File: Backend/src/main/java/com/example/ecommerce/dto/OAuthLoginRequest.java
package com.example.ecommerce.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class OAuthLoginRequest {

    @NotBlank(message = "Provider is required")
    private String provider;

    @NotBlank(message = "Provider user ID is required")
    private String providerUserId;

    @NotBlank(message = "Email is required")
    private String email;

    private String fullName;
    private String avatarUrl;
}

File: Frontend/lib/models/oauth_login_request.dart
class OAuthLoginRequest {
  final String provider;
  final String providerUserId;
  final String email;
  final String? fullName;
  final String? avatarUrl;

  OAuthLoginRequest({
    required this.provider,
    required this.providerUserId,
    required this.email,
    this.fullName,
    this.avatarUrl,
  });

  Map<String, dynamic> toJson() {
    return {
      'provider': provider,
      'providerUserId': providerUserId,
      'email': email,
      'fullName': fullName,
      'avatarUrl': avatarUrl,
    };
  }
}
